### PR TITLE
 feat(ts/ui): add finish detour button 

### PR DESCRIPTION
**Pull request by firestack**
This adds the finish button and sets up a stable area for the button "footer" and scroll area. This does not add tests or any functionality yet, because that depends on the creation of the share detours panel which requires the #2421 refactor.

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1206694541310031

### CODE REVIEW

File: assets/src/components/detours/diversionPanel.tsx [hunk 1]
```diff
@@ -11,6 +11,7 @@ export interface DiversionPanelProps {
   routeDescription: string
   routeOrigin: string
   routeDirection: string
+  onSubmit?: () => void
```
[the remark this answers is not in the document]
Comment (firestack) [reply assets/src/components/detours/diversionPanel.tsx:14]: I'll think on this, I'm not 100% sold but I'll get back to ya.

File: assets/src/components/detours/diversionPanel.tsx [hunk 2]
```diff
@@ -20,57 +21,66 @@ export const DiversionPanel = ({
...
+        {missedStops && (
+          <section className="pb-3">
+            <h2 className="c-diversion-panel__h3">Missed Stops</h2>
+            {missedStops}
+          </section>
+        )}
+      </Panel.Body.ScrollArea>
+
+      <Panel.Body.Footer>
+        <Button className="flex-grow-1 m-3" onClick={onSubmit}>
```
Comment (joshlarson) [inline assets/src/components/detours/diversionPanel.tsx:80]: IIRC - the button should only appear when the detour is completely finished, right? Is that slated for a follow-up, or is that going into this PR?

Comment (firestack) [reply assets/src/components/detours/diversionPanel.tsx:80]: oooh great catch, thanks!
I'll do that in this PR

**Comment by joshlarson** [comment]
> This does not add tests or any functionality yet, because that depends on the creation of the share detours panel which requires the https://github.com/mbta/skate/pull/2421 refactor.

Since ☝️ refactor is merged, does that mean that this PR can add in a test or two for the button now?

**Comment by firestack** [comment]
> > This does not add tests or any functionality yet, because that depends on the creation of the share detours panel which requires the #2421 refactor.
> 
> Since ☝️ refactor is merged, does that mean that this PR can add in a test or two for the button now?

😅 I meant to state that "depends on the creation of the share detours panel" is going to be another PR, that requires the #2421 PR.

But with https://github.com/mbta/skate/pull/2420#discussion_r1503255614 I will have tests!

File: assets/src/components/detours/diversionPanel.tsx (hunk 2 above)
Comment (firestack) [reply assets/src/components/detours/diversionPanel.tsx:82]: How's this? fc6b00d0de6a568af012ebcae6b8e0936b824691

File: assets/src/components/detours/diversionPanel.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (firestack) [reply assets/src/components/detours/diversionPanel.tsx]: 83adb5c63f268c1945acd7d9463fb54ce622b3d9

File: assets/src/components/detours/diversionPanel.tsx (hunk 2 above)
Comment (joshlarson) [reply assets/src/components/detours/diversionPanel.tsx]: Love it! Thanks!

**Review by joshlarson — APPROVED** [review]

**Pull request merged**